src/include/ipc/dai.h

```diff
@@ -88,4 +88,16 @@ struct sof_ipc_dai_config {
 	};
 } __attribute__((packed, aligned(4)));
 
+/* general purpose DAI clock control */
+struct sof_ipc_dai_clkctrl {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t type;		/**< DAI type - enum sof_ipc_dai_type */
+	uint32_t dai_index;	/**< index of this type dai */
+
+	/* HW specific data */
+	union {
```

Comment:
Dont think we need the union here as the interface looks generic.

Reply:
in fact, I would just put the
```c
	uint32_t clk_en;	/**&lt; Zero to disable the clock */
        uint32_t clk_id;	/**&lt; Clock id to operate */
```
in here directly

Reply:
fixed